redfish-core/lib/account_service.hpp

```diff
@@ -2129,11 +2326,14 @@ inline void requestAccountServiceRoutes(App& app)
             std::optional<std::string> roleId;
             std::optional<bool> locked;
             std::optional<nlohmann::json> oem;
+            std::optional<std::vector<std::string>> accountType;
+
+            bool isUserItself = (username == req.session->username);
```

Comment:
Should isUserItself be set after we check if req.session is a nullptr (as in line 2342)?